### PR TITLE
GO-3194 payments/nameservice protocol refactoring

### PR DESCRIPTION
**Pull request by AnthonyAkentiev**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: pkg/lib/pb/model/protos/models.proto [hunk 1]
```diff
@@ -1218,30 +1208,38 @@ message MembershipTierData {
...
     string colorStr = 11;
+
+    // Stripe platform-specific data:
+    string stripeProductId = 12;
+    string stripeManageUrl = 13;
+    uint32 stripePriceUsdCents = 14;
+
+    // iOS platform-specific data:
+    string iosProductId = 15;
+    string iosManageUrl = 16;
```
Comment (AnthonyAkentiev) [inline pkg/lib/pb/model/protos/models.proto:1240]: Or this field is not PER-tier?

File: pkg/lib/pb/model/protos/models.proto [hunk 2]
```diff
@@ -1218,30 +1208,38 @@ message MembershipTierData {
...
+
+    // Android platform-specific data:
+    string androidProductId = 17;
+    string androidManageUrl = 18;
```
Comment (AnthonyAkentiev) [inline pkg/lib/pb/model/protos/models.proto:1244]: Or this field is not PER-tier?

File: pkg/lib/pb/model/protos/models.proto [hunk 3]
```diff
@@ -1157,21 +1157,7 @@ message ManifestInfo {
...
-        // it will still create an active subscription, but with NO features
-        TierBuilder1WeekTEST = 2;
-        // this tier can be used just for testing in debug mode
-        // it will still create an active subscription, but with NO features
-        TierCoCreator1WeekTEST = 3;
-        TierBuilder = 4;
-        TierCoCreator = 5;
-    }
-    
+message Membership {
```
Comment (KirillSto) [inline pkg/lib/pb/model/protos/models.proto:1160]: Where are all these tiers identifiers are going? To any-sync protocol?

File: pkg/lib/pb/model/protos/models.proto [hunk 4]
```diff
@@ -1157,21 +1157,7 @@ message ManifestInfo {
...
-        // it will still create an active subscription, but with NO features
-        TierCoCreator1WeekTEST = 3;
-        TierBuilder = 4;
-        TierCoCreator = 5;
-    }
-    
+message Membership {    
     enum Status {
         StatusUnknown = 0;
         // please wait a bit more
```
Comment (konstantiniiv) [inline pkg/lib/pb/model/protos/models.proto:1163]: I don’t understand what it means "please wait a bit more"? wait on the client side for changing the status from StatusUnknown to another? there will be other statuses, I suppose?

File: pkg/lib/pb/model/protos/models.proto [hunk 5]
```diff
@@ -1186,12 +1172,16 @@ message Membership {
     }
     
     enum PaymentMethod {
-        MethodCard = 0; // Stripe
```
Comment (konstantiniiv) [inline pkg/lib/pb/model/protos/models.proto:1189]: maybe it's worth adding 'None' as the default value, for a situation when you can buy this Tier?

File: pkg/lib/pb/model/protos/models.proto [hunk 6]
```diff
@@ -1186,12 +1172,16 @@ message Membership {
...
-        MethodGooglePay = 3;
-        MethodAppleInapp = 4;
-        MethodGoogleInapp = 5;
+        MethodInappApple = 2;
+        MethodInappGoogle = 3;
+    }
+
+    enum EmailVerificationStatus {
+        NotVerified = 0;
+        CodeSent = 1;
```
Comment (konstantiniiv) [inline pkg/lib/pb/model/protos/models.proto:1183]: after sending the erroneous code, there will be a status "NotVerified"?

File: pkg/lib/pb/model/protos/models.proto (hunk 1 above)
Comment (ignatovv) [reply pkg/lib/pb/model/protos/models.proto:1240]: it is better to keep it per tier

File: pkg/lib/pb/model/protos/models.proto (hunk 3 above)
Comment (AnthonyAkentiev) [reply pkg/lib/pb/model/protos/models.proto:1160]: Yes, If you want/need -> I can keep it here also and roll this change back

File: pkg/lib/pb/model/protos/models.proto (hunk 4 above)
Comment (AnthonyAkentiev) [reply pkg/lib/pb/model/protos/models.proto]: How subscription purchase works (in Stripe case):

* User requests a new subscription.
* User is redirected to Stripe.
* User makes the payment.
* Stripe immediately returns "success" and redirects you via a deeplink <-- at this point, the user may still not have the subscription.
* (Uncertain when) Within a certain period of time, Stripe sends us a webhook <-- at this point, the subscription will be created and will enter the **pending** state.
* (Uncertain when) We process it and load various settings: names, restrictions, etc.
Once everything is completed, the subscription transitions to the active state.

added more comments to the .proto file

File: pkg/lib/pb/model/protos/models.proto (hunk 6 above)
Comment (AnthonyAkentiev) [reply pkg/lib/pb/model/protos/models.proto]: No. I have added more comments, please review.

File: pkg/lib/pb/model/protos/models.proto (hunk 3 above)
Comment (KirillSto) [reply pkg/lib/pb/model/protos/models.proto:1160]: No, it is better to keep it at any-sync, you are right

**Review by fat-fellow — APPROVED** [review]

**Pull request merged**